src/Tests/Microsoft.DotNet.ApiCompatibility.Tests/CustomSideNameTests.cs

```diff
@@ -0,0 +1,225 @@
+
+namespace CompatTests
+{
+  public class First { }
+}
+";
+
+            string rightSyntax = @"
+namespace CompatTests
+{
+  public class First
+  {
+    public string Method1() { }
+  }
+}
+";
+
+            ApiComparer differ = new();
+            bool enableNullable = false;
+            IAssemblySymbol left = SymbolFactory.GetAssemblyFromSyntax(leftSyntax, enableNullable);
+            IAssemblySymbol right = SymbolFactory.GetAssemblyFromSyntax(rightSyntax, enableNullable);
+            string expectedLeftName = "ref/net6.0/a.dll";
+            string expectedRightName = "lib/net6.0/a.dll";
+            differ.StrictMode = true;
+            IEnumerable<CompatDifference> differences = differ.GetDifferences(new[] { left }, new[] { right }, leftName: expectedLeftName, rightName: expectedRightName);
+            Assert.Single(differences);
+            AssertNames(differences.First(), expectedLeftName, expectedRightName, leftFirst: false);
+        }
+
+        [Fact]
+        public void MultipleRightsMetadataInformationIsUsedAsName()
+        {
+            string leftSyntax = @"
+namespace CompatTests
+{
+  public class First
+  {
+    public class FirstNested
+    {
+      public class SecondNested
+      {
+        public class ThirdNested
+        {
+          public string MyField;
+        }
+      }
+    }
+  }
+}
+";
+
+            string[] rightSyntaxes = new[]
+            { @"
+namespace CompatTests
+{
+  public class First
+  {
+    public class FirstNested
+    {
+      public class SecondNested
+      {
+        public class ThirdNested
+        {
+        }
+      }
+    }
+  }
+}
+",
+            @"
+namespace CompatTests
+{
+  public class First
+  {
+    public class FirstNested
+    {
+      public class SecondNested
+      {
+        public class ThirdNested
+        {
+        }
+      }
+    }
+  }
+}
+",
+            @"
+namespace CompatTests
+{
+  public class First
+  {
+    public class FirstNested
+    {
+      public class SecondNested
+      {
+        public class ThirdNested
+        {
+        }
+      }
+    }
+  }
+}
+"};
+
+            ApiComparer differ = new();
+            ElementContainer<IAssemblySymbol> left =
+                new(SymbolFactory.GetAssemblyFromSyntax(leftSyntax), new MetadataInformation(string.Empty, string.Empty, "ref/net6.0/a.dll"));
+
+            IList<ElementContainer<IAssemblySymbol>> right = SymbolFactory.GetElementContainersFromSyntaxes(rightSyntaxes);
+
+            IEnumerable<(MetadataInformation, MetadataInformation, IEnumerable<CompatDifference>)> differences =
+                differ.GetDifferences(left, right);
+
+            int i = 0;
+            foreach ((MetadataInformation, MetadataInformation, IEnumerable<CompatDifference> differences) diff in differences)
+            {
+                Assert.Single(diff.differences);
+                AssertNames(diff.differences.First(), left.MetadataInformation.AssemblyId, right[i++].MetadataInformation.AssemblyId);
```

Comment:
what will be the value of left and right assemblyId ? will it be the default value ?

Reply:
No, `left.MetadataInformation.AssemblyId` is set in this test in L191: https://github.com/dotnet/sdk/pull/18533/files#diff-3aba641bc1a70ca2e3c5f1781db2e432c9e17665dd52d91a5f2de0214bf4bf7fR191

`right.MetadataInformation.AssemblyId` will be `runtime-n`: https://github.com/dotnet/sdk/blob/f2b190a84a31e7f836129b6c6524974af2e5bbc2/src/Tests/Microsoft.DotNet.ApiCompatibility.Tests/SymbolFactory.cs#L37